source/codegen/templates/service_helpers.mako

```diff
@@ -7,6 +7,8 @@
 <%def name="define_init_method_body(function_name, function_data, parameters)">\
```

Comment:
The updates here can be simplified a bit. If you look at the [caller of this def in service.cpp.mako](https://github.com/doshirohan/grpc-device/blob/features/ni-dmm/source/codegen/templates/service.cpp.mako#L77) the function_data is equivalent to the `function[function_name]` you use here and `functions[method_name]` used in `has_close_function`.

You can remove the `function` variable you added and pass `function_data` to `has_close_function`. Also on 25, you can use `function_data` instead of `function[function_name]`.